samples/volumes/example5.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import kfp.dsl as dsl
+
+
+@dsl.pipeline(
+    name="Example 5",
+    description="The fifth example of the design doc."
+)
+def example5(rok_url):
+    vol1 = dsl.PipelineVolume(
+        name="vol1",
+        size="1Gi",
+        annotations={"rok/origin": rok_url},
+        mode=dsl.VOLUME_MODE_RWM
+    )
+
+    step1 = dsl.ContainerOp(
+        name="step1_concat",
+        image="library/bash:4.4.23",
+        command=["sh", "-c"],
+        arguments=["cat /data/file*| gzip -c >/data/full.gz"],
+        volumes={"/data": vol1}
+    )
+
+    step1_snap = dsl.PipelineVolumeSnapshot(
+        step1.volume,
+        name="snap1"
+    )
+
+    vol2 = dsl.PipelineVolume(
+        data_source=step1_snap,
+        name="vol2"
+    )
+
+    step2 = dsl.ContainerOp(
+        name="step2_gunzip",
+        image="library/bash:4.4.23",
+        command=["gunzip", "-k", "/data/full.gz"],
+        volumes={"/data": vol2}
+    )
+
+    step2_snap = dsl.PipelineVolumeSnapshot(
```

Comment:
Since this feature is still in Alpha, I am worried that not every user can use it. Can we not specialize it in DSL for now and just use k8s spec to represent it and we provide a K8sResourceOp to support to create it in DSL?

Reply:
@hongye-sun 
> Since this feature is still in Alpha, I am worried that not every user can use it. Can we not specialize it in DSL for now and just use k8s spec to represent it and we provide a K8sResourceOp to support to create it in DSL?

We agree with creating a special `Op` to create resources.
But using the k8s resource manually does not work, it leads to lots of boilerplate.
Using a simple type in the DSL that inherits from this resource simplifies the pipelines significantly.

This is also what has gathered very positive end user feedback, both in the design doc issue #801, and in discussions we've had with potential users in the wild. So, I think it is best to keep it. We will add a warning that this will not work unless supported by the cluster.